### PR TITLE
[embedlite-components] Trim spurious entries from addhistory. Contributes to JB#55325 OMP#JOLLA-326

### PR DESCRIPTION
**Pull request by chriadam**
When reloading a session, the "links" array in the embedui:addhistory
message will contain two spurious entries:
- the about:blank entry for the preload view
- the most recent loaded URI (which will match the current loaded uri)
These two entries need to be trimmed, to ensure that after session
reloading is complete, history navigation works as expected.

### CODE REVIEW

**Comment by chriadam** [comment]
@rainemak @mkenttala @llewelld This seems a bit like fixing the symptom rather than the cause, but it seems to resolve the issue according to my testing.

File: jsscripts/embedhelper.js [hunk 1]
```diff
@@ -245,6 +246,19 @@ EmbedHelper.prototype = {
         // Adjust index to the range max session history entries.
         let index = Math.max(0, (aMessage.data.index - itemsToRemove));
         links.splice(0, itemsToRemove);
+        // If we are reloading a previous session, the links will include
+        // an about:blank page and the currently loading page.  Remove those two.
+        let removedSpurious = false;
+        if (originalCount === 0 && links[links.length - 1] == "about:blank") {
```
Comment (rainemak) [inline jsscripts/embedhelper.js:252]: Hmmp... so "about:blank" gets there into the history. Sound that it's getting to the database as well.

The links are build here:
https://github.com/sailfishos/sailfish-browser/blob/next/apps/qtmozembed/declarativewebpage.cpp#L174 (see also DeclarativeWebPage::onTabHistoryAvailable)

History entries added here:
https://github.com/sailfishos/sailfish-browser/blob/next/apps/storage/dbworker.cpp#L446

Wondering if we should this cleanup on the DeclarativeWebPage::restoreHistory() instead. Reasoning being that here we add whatever is requested to the history but in restoryHistory we construct previous tab history after tab history is available.

**Comment by rainemak** [comment]
First entry in the restored history seems to be still about:blank. I'd not start cleaning up random about:blanks in between of start and end but cleaning up first and last would be good idea, I think.

File: jsscripts/embedhelper.js (hunk 1 above)
Comment (rainemak) [reply jsscripts/embedhelper.js:252]: No need to move this to the webpage.cpp. Let's handle here.

Noticed that first entry (oldest entry) is also about:blank. By following your steps:

1. Open browser
2. Navigate to jolla.com
3. Navigate to sailfishos.org
4. Navigate to github.com/sailfishos
5. Close browser
6. Reopen browser
7. github.com/sailfishos loads
8. Navigate back to load sailfishos.org
9. Navigate back to load jolla.com
10. Navigate back button still active... press it => about:blank

**Review by rainemak — CHANGES_REQUESTED** [review]
Let's splice also oldest entry if that happens to be about:blank

File: jsscripts/embedhelper.js (hunk 1 above)
Comment (chriadam) [reply jsscripts/embedhelper.js:253]: Indeed.  Added a fix for this too, but even with this fix it's still not perfect:

Steps:
1) rm -rf .local/share/org.sailfishos/browser/
2) open browser --> navigate to jolla.com and then sailfishos.org.  close browser
3) open browser --> browser will load sailfishos.org, press back to go back to jolla.com, note that back button vanishes as wanted - all SEEMS well.  close browser.
4) open browser --> the links looks like this:

```
XXXXXXXXXXX links prior to fix: 
XXXXXXXXXx -> 0 = about:blank
XXXXXXXXXx -> 1 = https://jolla.com/
XXXXXXXXXx -> 2 = https://sailfishos.org/
XXXXXXXXXx -> 3 = about:blank
XXXXXXXXXx -> 4 = https://jolla.com/
XXXXXXXXXx -> 5 = about:blank
XXXXXXXXXXX links after fix: 
XXXXXXXXXx -> 0 = https://jolla.com/
XXXXXXXXXx -> 1 = https://sailfishos.org/
XXXXXXXXXx -> 2 = about:blank
```
I think we need to find out why the "about:blank" is being injected for each initial page-load and prevent it to begin with..

Comment (chriadam) [reply jsscripts/embedhelper.js:253]: Evening adding another fix like:
```
            // remove trailing about:blank page due to back-navigation
            if (links.length > 0 && links[links.length - 1] == "about:blank") {
                links.splice(links.length-1, 1);
                index -= 1;
            }
```
just prior to the "remove leading about:blank" case, doesn't resolve the issue entirely: multiple rounds of open browser + close browser still causes the "about:blank" to be injected and not removed in this case, with output like:
```
XXXXXXXXXXX links prior to fix: 
XXXXXXXXXx -> 0 = about:blank
XXXXXXXXXx -> 1 = https://jolla.com/
XXXXXXXXXx -> 2 = https://sailfishos.org/
XXXXXXXXXx -> 3 = about:blank
XXXXXXXXXx -> 4 = https://jolla.com/
XXXXXXXXXx -> 5 = about:blank
XXXXXXXXXx -> 6 = https://jolla.com/
XXXXXXXXXx -> 7 = about:blank
XXXXXXXXXXX links after fix: 
XXXXXXXXXx -> 0 = https://jolla.com/
XXXXXXXXXx -> 1 = https://sailfishos.org/
XXXXXXXXXx -> 2 = about:blank
XXXXXXXXXx -> 3 = https://jolla.com/
```

Comment (chriadam) [reply jsscripts/embedhelper.js:253]: I'll take a look at DeclarativeWebPage m_restoredTabHistory things.

Comment (chriadam) [reply jsscripts/embedhelper.js:253]: After a bunch of spelunking, I think this is the better solution: https://github.com/sailfishos/qtmozembed/pull/12/files

**Pull request closed without merging**